python/pyspark/sql/dataframe.py

```diff
@@ -2095,9 +2095,11 @@ def toPandas(self):
                         _check_dataframe_localize_timestamps
                     import pyarrow
 
-                    tables = self._collectAsArrow()
-                    if tables:
-                        table = pyarrow.concat_tables(tables)
+                    # Collect un-ordered list of batches, and list of correct order indices
+                    batches, batch_order = self._collectAsArrow()
+                    if batches:
```

Comment:
Not a big deal at all and personal preference: I would do this like `len(batches) > 0`.

Reply:
Sure, I was playing around with this being an iterator, but I will change it since it is a list now